wallet/wallet-cli/src/config.rs

```diff
@@ -0,0 +1,150 @@
+
+#[derive(clap::ValueEnum, Clone, Debug)]
+pub enum Network {
+    Mainnet,
+    Testnet,
+    Regtest,
+    Signet,
+}
+
+#[derive(Parser, Debug)]
+pub struct WalletCliArgs {
+    /// Network
+    #[arg(long, value_enum, default_value_t = Network::Mainnet)]
+    pub network: Network,
+
+    /// Optional path to the directory containing the wallet files
+    #[clap(long)]
+    pub wallets_dir: Option<PathBuf>,
+
+    /// Optional wallet file name
+    #[clap(long)]
+    pub wallet_name: Option<PathBuf>,
+
+    /// Optional RPC address
+    #[clap(long)]
+    pub rpc_address: Option<SocketAddr>,
+
+    /// Path to the RPC cookie file. If not set, the value is read from the default cookie file location.
+    #[clap(long)]
+    pub rpc_cookie_file: Option<String>,
+
+    /// RPC username (either provide a username and password, or use a cookie file. You cannot use both)
+    #[clap(long)]
+    pub rpc_username: Option<String>,
+
+    /// RPC password (either provide a username and password, or use a cookie file. You cannot use both)
+    #[clap(long)]
+    pub rpc_password: Option<String>,
+
+    /// vi input mode
+    #[clap(long)]
+    pub vi_mode: bool,
+}
+
+#[derive(Debug)]
+pub struct WalletCliConfig {
+    pub chain_type: ChainType,
+    pub data_dir: PathBuf,
+    pub wallet_file: PathBuf,
+    pub rpc_address: SocketAddr,
+    pub rpc_username: String,
+    pub rpc_password: String,
+    pub vi_mode: bool,
+}
+
+const DEFAULT_WALLETS_DIR: &str = "wallets";
+const DEFAULT_WALLET_NAME: &str = "wallet";
+
+const COOKIE_FILENAME: &str = ".cookie";
+
+fn load_cookie(path: impl AsRef<Path>) -> Result<(String, String), WalletCliError> {
+    let content = std::fs::read_to_string(path.as_ref())
+        .map_err(|e| WalletCliError::CookieFileReadError(path.as_ref().to_owned(), e))?;
+    let (username, password) = content.split_once(':').ok_or(WalletCliError::InvalidConfig(
+        format!("Invalid cookie file {:?}: ':' not found", path.as_ref()),
+    ))?;
+    Ok((username.to_owned(), password.to_owned()))
+}
+
+impl WalletCliConfig {
+    pub fn from_args(args: WalletCliArgs) -> Result<WalletCliConfig, WalletCliError> {
+        let WalletCliArgs {
+            network,
+            wallets_dir,
+            wallet_name,
+            rpc_address,
+            rpc_cookie_file,
+            rpc_username,
+            rpc_password,
+            vi_mode,
+        } = args;
+
+        let chain_type = match network {
+            Network::Mainnet => ChainType::Mainnet,
+            Network::Testnet => ChainType::Testnet,
+            Network::Regtest => ChainType::Regtest,
+            Network::Signet => ChainType::Signet,
+        };
+
+        let data_dir = prepare_data_dir(
+            || default_data_dir_for_chain(chain_type.name()).join(DEFAULT_WALLETS_DIR),
+            &wallets_dir,
+        )
+        .map_err(WalletCliError::PrepareData)?;
+        let wallet_file = data_dir.join(wallet_name.unwrap_or(DEFAULT_WALLET_NAME.into()));
```

Comment:
The wallet data should not be related to the node data dir. The way it should be seen, is that the node is a stateful entity that keeps running no matter what, and an unbounded number of wallets can connect to it. The files of the wallets have nothing to do with the node. They can be on removable drives, on a network, or whatever.

We have a design here different that what bitcoin core does, which is something that has caused issues for Bitcoin. I prefer the way Monero does this, where the wallet is completely separate from the wallet, and the only thing that they make them communicate is the RPC interface, which is for us not necessarily just the RPC interface, but the NodeInterface trait in general.

Reply:
Sure, I understand that. But we still need a default directory to store the wallet files somewhere. I decided to use  the "wallets" directory in the default data dir (for example, "$HOME/.mintlayer/wallets" on Linux). It does not matter if the node is started or not on the same host.

Reply:
Please, no default dir/path for the wallet. This is wrong.